exla/test/exla/defn_expr_test.exs

```diff
@@ -1134,6 +1134,50 @@ defmodule EXLA.DefnExprTest do
     end
   end
 
+  describe "scatter_add" do
+    defn scatter_add(t, i, u) do
+      Nx.scatter_add(t, i, u)
+    end
+
+    test "scatter_add works for multi-dim tensor" do
+      target =
+        Nx.broadcast(0, {2, 3, 4})
+
+      indices = Nx.tensor([[[
```

Comment:
@josevalim @jonatanklosko at first I was going to support indices like the ones resulting from argsort, which are along an specific axis. However, this would hinder any progress with using this as the grad for gather.

Do you think this interface is fine? The shape of the indices doesn't matter much, we can think of it as something like iodata being converted to a binary, as the second test shows.

The only restriction is that the indices rank must be the same as the target rank + 1, since the last dim is filled with a tensor of shape {rank}.

Reply:
If this is ok, I'll move on to implementing the binary backend for this function

Reply:
I'd consider restricting the shape to be 2d, since logically any other shape isn't relevant.

Reply:
Similarly to gather, scatter could have versions working with larger slices, thought whatever route we go I think our `scatter` and `gather` should do *the same* ^^

Reply:
I see, so we expose, for instance, indices and updates like: `[[0, 1, 1], [0, 2, 3], [1, 1, 2]]` and `[1, 2, 3, 4]` and just reshape them accordingly at the EXLA level

Reply:
WDYT?

Reply:
Yeah, that's what I thought, but could definitely use another opinion :) @seanmor5, @josevalim